src/gpu/nvidia/cudnn_inner_product.cpp

```diff
@@ -72,23 +67,23 @@ status_t cudnn_inner_product_fwd_t::execute(const exec_ctx_t &ctx) const {
 
             std::vector<void *> args;
 
-            args.push_back(sc.memory<void *>(ih, src_acc));
-            args.push_back(sc.memory<void *>(ih, wei_acc));
-            args.push_back(
-                    ((pd()->with_bias()) ? sc.memory<void *>(ih, *bias_acc)
-                                         : nullptr));
-            args.push_back(sc.memory<void *>(ih, dst_acc));
+            args.push_back(arg_src.get_native_pointer(ih));
+            args.push_back(arg_wei.get_native_pointer(ih));
+            args.push_back(arg_bias.get_native_pointer(ih));
+            args.push_back(arg_dst.get_native_pointer(ih));
+
             args.push_back((pd()->inner_product_impl_->ip_using_scratchpad()
-                            ? sc.memory<void *>(ih, *ip_scratch_acc)
-                            : nullptr));
-            args.push_back((
-                    pd()->inner_product_impl_->need_to_transform_filter()
-                            ? sc.memory<void *>(ih, *spacial_scratch_acc)
+                            ? arg_ip_scratch.get_native_pointer(ih)
```

Comment:
Since we pass `nullptr` here anyway then let's just do  `arg_ip_scratch.get_native_pointer(ih)`.